kotlin-mbedtls/src/main/kotlin/org/opencoap/ssl/transport/DtlsServer.kt

```diff
@@ -132,9 +132,18 @@ class DtlsServer(
 
     fun numberOfSessions(): Int = executor.supply { sessions.size }.join()
 
-    fun setSessionAuthenticationContext(adr: InetSocketAddress, authenticationContext: String): CompletableFuture<Boolean> =
+    fun setSessionAuthenticationContext(adr: InetSocketAddress, authenticationCredentials: String?): CompletableFuture<Boolean> =
+        updateSessionAuthenticationContext(adr, DtlsSessionContext.DEFAULT_AUTH_CTX_LABEL, authenticationCredentials)
+
+    fun updateSessionAuthenticationContext(adr: InetSocketAddress, label: String, authenticationCredentials: String?): CompletableFuture<Boolean> =
```

Comment:
How about naming it:
```
fun putSessionAuthenticationContext(adr: InetSocketAddress, key: String, contextValue: String?)
```

Reply:
My idea was that the whole map is the context which keeps different authentication credentials associated with the dtls session during its lifetime. And with this method we are updating different keys of the context. In my opinion the naming is more consistent this way.

Reply:
That was my idea with a name, in a map classes, there is usually methods named `put`